### DISCUSSION TITLE
K-core data filtering with infinity loop?

### DISCUSSION THREAD

**Discussion by rowedenny** [RUCAIBox/RecBole#737] [General] [answered]
In the current implementation of [data filter by interaction number](https://github.com/RUCAIBox/RecBole/blob/1a5f66118c21078eabd4afc1d0ded87b7b50e04d/recbole/data/dataset/dataset.py#L637), it iteratively drops the items and users based on the filter conditions. 

Yet to my best knowledge, typically this dropout is only applied one-round instead of multiple round, otherwise, the volume of the original dataset will significantly decrease since one-round of drops will lead to some new "low frequent items and users"

Any reasons why we have an [infinity loop](https://github.com/RUCAIBox/RecBole/blob/1a5f66118c21078eabd4afc1d0ded87b7b50e04d/recbole/data/dataset/dataset.py#L637) here?

**Comment by hyp1231** [comment, marked as answer]
In RecBole, "k-core" has a similar meaning as "[Degeneracy](https://en.wikipedia.org/wiki/Degeneracy_(graph_theory))" in graph theory.

Many studies have followed this concept, one example is [Amazon Review Data (2018)](https://nijianmo.github.io/amazon/index.html#subsets).
> K-cores (i.e., dense subsets): These data have been reduced to extract the k-core, such that each of the remaining users and items have k reviews each.

Thus, k is the number of interaction records of **remaining** users and items, and we have an infinity loop here.

**Reply by rowedenny** [reply]
I see, thanks!
